### ISSUE TITLE
[a11y]: tooltip's behaviour fails WCAG in two ways

### ISSUE THREAD

**Issue by mbgower** [carbon-design-system/carbon#16152]
### Package

@carbon/react

### Browser

Chrome

### Operating System

MacOS

### Package version

@1.54.0

### React version

https://react.carbondesignsystem.com/?path=/story/components-tooltip--default

### Automated testing tool and ruleset

n/a

### Assistive technology

n/a

### Description

There are two issues with the current tooltip behaviour:
1. It should be dismissible with Esc, even if activated by hover
2. If I move the pointer over top of the exposed tooltip, it should not go away.

These are described in https://www.w3.org/WAI/WCAG21/Understanding/content-on-hover-or-focus.html as the Dismissible and Hoverable requirements.

### WCAG 2.1 Violation

Content on Hover or Focus (Level AA)

### Reproduction/example

https://react.carbondesignsystem.com/?path=/story/components-tooltip--default

### Steps to reproduce

1. Got to https://react.carbondesignsystem.com/?path=/story/components-tooltip--default
2. Hover over the i icon
3. Press Esc. Nothing happens. Expected behaviour is the tooltip should disappear. 
4. With the tooltip still visible, move the pointer so it is over top of the tooltip. It disappears. Expected behaviour is the tooltip should remain visible so long at the pointer is over either the target of the tooltip (so that a low vision user can move their magnification to it and read the tip)

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by tay1orjones** [comment]
> It should be dismissible with Esc, even if activated by hover

Yeah I don't recall this being part of the spec for Tooltip's updates for v11, this should be pretty easy to add. 

> If I move the pointer over top of the exposed tooltip, it should not go away.

This is a regression, it _used_ to work this way, [here's an example from v11.25.0](https://deploy-preview-13363--v11-carbon-react.netlify.app/?path=/story/components-tooltip--default). I just grabbed that deploy preview at random from the [list of release PRs](https://github.com/carbon-design-system/carbon/pulls?page=7&q=is%3Apr+chore%28release%29+is%3Aclosed). We could probably track down exactly what version it stopped working to get an sense of why/how this regression was introduced and how to fix it.
